### PR TITLE
GH-1883: Move test-only Turtle parser generator

### LINKED ISSUE #1883: Move generator for the jena-core legacy Turtle parser to testing/
### Change

Move the Turtle parser used only for test data in jena-core so that it is not at the top level of `jena-core`.

Update to make sure it can regenerate a parser.

This parser is not available to `main/java` - it exists only so Turtle data in long-standing JUnit tests can be read. 

### Are you interested in contributing a pull request for this task?

Yes

### PR DESCRIPTION
**Pull request by afs**
GitHub issue resolved #1883 

Pull request Description:
Clarification motivated by #1882.

----

 - [x] Key commit messages start with the issue number (GH-xxxx or JENA-xxxx)

By submitting this pull request, I acknowledge that I am making a contribution to the Apache Software Foundation under the terms and conditions of the [Contributor's Agreement](https://www.apache.org/licenses/contributor-agreements.html).

### CODE REVIEW

File: jena-core/testing/ttl_test/README [hunk 1]
```diff
@@ -1,4 +1,4 @@
-This is a Turtle parser used in support of jena-core tests.
+This is a Turtle parser used in support of jena-core tests and nothign more.
```
Comment (kinow) [inline jena-core/testing/ttl_test/README:1]: s/nothign/nothing

File: jena-core/src/test/java/org/apache/jena/ttl_test/turtle/parser/JavaCharStream.java [hunk 2]
```diff
@@ -729,7 +730,7 @@ public void Done()
    * @param newLine the new line number.
    * @param newCol the new column number.
    */
-
+
```
Comment (kinow) [inline jena-core/src/test/java/org/apache/jena/ttl_test/turtle/parser/JavaCharStream.java:733]: Not sure if GitHub UI, or if these whitespaces were added intentionally :+1:

**Review by kinow — APPROVED** [review]
Couple comments, no blockers. :+1:

File: jena-core/testing/ttl_test/README (hunk 1 above)
Comment (afs) [reply jena-core/testing/ttl_test/README:1]: Done.

File: jena-core/src/test/java/org/apache/jena/ttl_test/turtle/parser/JavaCharStream.java (hunk 2 above)
Comment (afs) [reply jena-core/src/test/java/org/apache/jena/ttl_test/turtle/parser/JavaCharStream.java:733]: This is automatically generated by different versions of javacc.

(re-running the generator, same version, gives no changes which is helpful)

**Pull request merged**